core/src/main/scala/cats/syntax/flatMap.scala

```diff
@@ -50,3 +52,17 @@ final class IfMOps[F[_]](val fa: F[Boolean]) extends AnyVal {
    */
   def ifM[B](ifTrue: => F[B], ifFalse: => F[B])(implicit F: FlatMap[F]): F[B] = F.ifM(fa)(ifTrue, ifFalse)
 }
+
+  /**
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   *
+   * scala> val a: Int = 10
+   * scala> a.tailRecM[Option,String](i => if (i == 20) Some(Right("done")) else Some(Left(i+1)))
+   * res0: Option[String] = Some(done)
+   *}}}
+   */
+final class FlatMapIdOps[A](val a: A) extends AnyVal {
+  def tailRecM[F[_],B](f: A => F[Either[A,B]])(implicit F: FlatMap[F]): F[B] = F.tailRecM(a)(f)
+}
```

Comment:
Thanks, @aaronlevin!

Minor: it might be more helpful to have this scaladoc comment on the `tailRecM` method instead of the `FlatMapIdOps` class. If someone hovers over the `tailRecM` in something like `10.tailRecM(...)` in their IDE, this scaladoc might pop up if it's defined on the method as opposed to the class. Also conceivably there could be more methods that we add to this class.